Doc/whatsnew/3.12.rst

```diff
@@ -692,6 +692,11 @@ Deprecated
   Python 3.14, when ``'data'`` filter will become the default.
   See :ref:`tarfile-extraction-filter` for details.
 
+* The bitwise inversion operator (~) on bool is deprecated. It will throw an
+  error in Python 3.14. Use ``not`` for logical negation of bools instead.
+  In the rare case that you really need the bitwise inversion of the underlying
+  ``int``, convert to int explicitly ``~int(x)``. (Contributed by Tim Hoffmann
```

Comment:
```suggestion
* The bitwise inversion operator (``~``) on bool is deprecated. It will throw an
  error in Python 3.14. Use ``not`` for logical negation of bools instead.
  In the rare case that you really need the bitwise inversion of the underlying
  ``int``, convert to int explicitly with ``~int(x)``. (Contributed by Tim Hoffmann
```